clients/android/app/src/main/java/app/lockbook/screen/ShareReceiverActivity.kt

```diff
@@ -0,0 +1,198 @@
+                if (receivedUri != null) {
+                    uris.add(receivedUri)
+                }
+            }
+        }
+
+        binding.importButton.setOnClickListener {
+            binding.importProgress.visibility = View.VISIBLE
+            binding.importButton.isEnabled = false
+            lifecycleScope.launch {
+                val fileId = importFromUris()
+                importedFileId.value = fileId
+            }
+        }
+
+        val sharedFilesCount = uris.count()
+        val subTitle = if (sharedFilesCount > 1) {
+            "Importing $sharedFilesCount files"
+        } else if (sharedFilesCount == 1) {
+            "Importing " + getUriFileName(uris[0])
+        } else {
+            ""
+        }
+        binding.toolbar.setSubtitle(subTitle)
+
+        supportFragmentManager.registerFragmentLifecycleCallbacks(
+            fragmentFinishedCallback,
+            false
+        )
+
+        binding.toolbar.setOnMenuItemClickListener {
+            activityModel.launchTransientScreen(TransientScreen.Create(model.currentParent.id))
+            CreateFileDialogFragment().show(
+                supportFragmentManager,
+                CreateFileDialogFragment.TAG
+            )
+            true
+        }
+
+        importedFileId.observe(this) { id ->
+            binding.importProgress.visibility = View.INVISIBLE
+            binding.importButton.isEnabled = true
+
+            if (id !== null) {
+                startActivity(
+                    Intent(this, MainScreenActivity::class.java).apply {
+                        putExtra(IMPORTED_FILE_KEY, id)
+                    }
+                )
+            }
+        }
+    }
+
+    private fun getUriFileName(uri: Uri): String? {
+        var fileName = "untitled"
+        contentResolver.query(uri, null, null, null, null)?.use { cursor ->
+            if (cursor.moveToFirst()) {
+                val displayNameIndex = cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME)
+
+                if (displayNameIndex != -1) {
+                    fileName = cursor.getString(displayNameIndex)
+                }
+            }
+        }
+        return fileName
+    }
+
+    private fun importFromUris(): String? {
+        var newFileId: String? = null
+
+        for (uri in uris) {
+            try {
+                val data =
+                    contentResolver.openInputStream(uri)?.use { stream -> stream.readBytes() }
+                val lbFile = Lb.createFile(getUriFileName(uri), model.currentParent.id, true)
+                newFileId = lbFile.id
+                Lb.writeDocumentBytes(lbFile.id, data)
+            } catch (err: LbError) {
+                alertModel.notifyError(err)
+            } catch (err: Exception) {
+                err.message?.let { alertModel.notify(it) }
+            }
+        }
+
+        // todo: when android supports multiple tabs. open all the imported files?
+        return newFileId
+    }
```

Comment:
Lockbook api has a function for importing, we should be using that instead of creating and writing our own files

Reply:
Good point, on other platforms i've opted to only support URLS until lb supports it

Reply:
Yeah, not by directly passing the URI, but envisioning we can handle URI conversions, and pass in the raw info to rust. That doesn't cover all cases, but seems possible with something like a remote server. All in all, my philosophy is to use lb whenever I can, and not come up with too many bespoke platform solutions to stuff like this (because we're going to have to duplicate it across platforms). But aligned with this - not too concerned